api/v1alpha1/recipe_types.go

```diff
@@ -70,15 +75,34 @@ type Group struct {
 	// Whether to include any cluster-scoped resources. If nil or true, cluster-scoped resources are
 	// included if they are associated with the included namespace-scoped resources
 	IncludeClusterResources *bool `json:"includeClusterResources,omitempty"`
+	// Selects namespaces by label
+	IncludedNamespacesByLabel *metav1.LabelSelector `json:"includedNamespacesByLabel,omitempty"`
 	// List of namespaces to include.
 	//+optional
 	IncludedNamespaces []string `json:"includedNamespaces,omitempty"`
+	// List of namespace to exclude
+	ExcludedNamespaces []string `json:"excludedNamespaces,omitempty"`
```

Comment:
Can we also add excludedNamespacedByLabel?

Reply:
We thought about it, but given that other consumers haven't implemented a `excludedNamespacesByLabel` it could be that there's no demand for it, or that there exist other complications that are not immediately apparent before attempting implementation. Given that it's always easier to add rather than remove fields from an API, it feels prudent to leave it out until there's an explicit request for it.

Reply:
In this PR I am only trying to match the CRD with other consumers expectations. We can add other features in a subsequent PR.